src/app/components/frequency-picker/frequency-picker.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatChip } from '@angular/material/chips';
+
+enum Frequency {
+  daily = 'daily',
+  weekly = 'weekly',
+  monthly = 'monthly',
+}
+
+enum DayOfMonth {
+  first = 'First',
+  last = 'Last',
+  specificLast = 'SpecificLast',
+  custom = 'Custom',
+}
+
+@Component({
+  selector: 'app-frequency-picker',
+  templateUrl: './frequency-picker.component.html',
+  styleUrls: ['./frequency-picker.component.scss'],
+})
+export class FrequencyPickerComponent implements OnInit {
+  frequencyEnum = Frequency;
+  DayOfMonthEnum = DayOfMonth;
+  today: Date = new Date();
+
+  daysOfWeek: Map<number, string> = new Map([
+    [0, 'Sunday'],
+    [1, 'Monday'],
+    [2, 'Tuesday'],
+    [3, 'Wednesday'],
+    [4, 'Thursday'],
+    [5, 'Friday'],
+    [6, 'Saturday'],
+  ]);
+
+  frequencyPicker = new FormGroup({
+    startDateControl: new FormControl(this.today, Validators.required),
+    timeControl: new FormControl('08:00', Validators.required),
```

Comment:
First parameter should be passed by variable so you can create variable and assign values